### PR TITLE
*: Update the return value type of `CutRow` and `CutIndexKey` to `[][]byte`

### PR DESCRIPTION
**Pull request by zimulala**
tablecodec: Update the return value type of `CutRow` and `CutIndexKey` to `[][]byte`
store: Rollback old code
store: Update the interface

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @coocood @shenli

File: store/tikv/mock-tikv/cop_handler_dag.go [hunk 1]
```diff
@@ -119,15 +114,11 @@ func (h *rpcHandler) buildExec(ctx *dagContext, curr *tipb.Executor) (executor,
...
-		ids := make([]int64, 0, len(columns))
-		for _, col := range columns {
-			ids = append(ids, col.GetColumnId())
-		}
 		ranges := h.extractKVRanges(ctx.keyRanges, *curr.IdxScan.Desc)
 		currExec = &indexScanExec{
 			IndexScan:   curr.IdxScan,
 			kvRanges:    ranges,
-			ids:         ids,
+			cols:        len(columns),
```
Comment (hanfei1991) [inline store/tikv/mock-tikv/cop_handler_dag.go]: rename cols to colsLen

File: store/tikv/mock-tikv/executor.go [hunk 2]
```diff
@@ -313,37 +334,37 @@ func handleRowData(columns []*tipb.ColumnInfo, colTps map[int64]*types.FieldType
...
 	}
 
 	return values, nil
 }
 
-// setColumnValueToEval puts column values into evaluator, the values will be used for expr evaluation.
-func setColumnValueToEval(eval *xeval.Evaluator, handle int64, row map[int64][]byte, cols map[int64]*tipb.ColumnInfo) error {
-	for colID, col := range cols {
+// setSchemaValueToEval puts column values into evaluator, the values will be used for expr evaluation.
+func setSchemaValueToEval(eval *xeval.Evaluator, handle int64, row [][]byte, cols []*tipb.ColumnInfo, colIDs map[int64]int) error {
```
Comment (zimulala) [inline store/tikv/mock-tikv/executor.go]: The columns count of the last argument is less than or equal to columns count in `cols`.

**Comment by zimulala** [comment]
PTAL @shenli

File: store/tikv/mock-tikv/executor.go [hunk 3]
```diff
@@ -30,12 +30,12 @@ import (
...
 type executor interface {
 	SetSrcExec(executor)
-	Next() (int64, map[int64][]byte, error)
+	Next() (int64, [][]byte, error)
 }
 
 type tableScanExec struct {
 	*tipb.TableScan
-	colTps      map[int64]*types.FieldType
+	schema      map[int64]int
```
Comment (coocood) [inline store/tikv/mock-tikv/executor.go]: Why not just name it `colIDs`?

File: tablecodec/tablecodec.go [hunk 4]
```diff
@@ -298,7 +298,48 @@ func DecodeRow(b []byte, cols map[int64]*types.FieldType) (map[int64]types.Datum
 	return row, nil
 }
 
-// CutRow cut encoded row into byte slices and return interested columns' byte slice.
+// CutRowNew cuts encoded row into byte slices and return columns' byte slice.
+// Row layout: colID1, value1, colID2, value2, .....
+func CutRowNew(data []byte, cols map[int64]int) ([][]byte, error) {
```
Comment (coocood) [inline tablecodec/tablecodec.go]: s/cols/colIDs

File: store/tikv/mock-tikv/executor.go [hunk 5]
```diff
@@ -313,37 +325,37 @@ func handleRowData(columns []*tipb.ColumnInfo, colTps map[int64]*types.FieldType
 			if err1 != nil {
 				return nil, errors.Trace(err1)
 			}
-			values[col.GetColumnId()] = handleData
+			values[colIDs[id]] = handleData
```
Comment (coocood) [inline store/tikv/mock-tikv/executor.go]: Define a `colOffset` at the beginning of the loop is more readable.

File: store/tikv/mock-tikv/executor.go [hunk 6]
```diff
@@ -313,37 +325,37 @@ func handleRowData(columns []*tipb.ColumnInfo, colTps map[int64]*types.FieldType
...
 			continue
 		}
 		if len(col.DefaultVal) > 0 {
-			values[col.GetColumnId()] = col.DefaultVal
+			values[colIDs[id]] = col.DefaultVal
 			continue
 		}
 		if mysql.HasNotNullFlag(uint(col.GetFlag())) {
-			return nil, errors.New("Miss column")
+			return nil, errors.Errorf("Miss column %d", col.GetColumnId())
```
Comment (coocood) [inline store/tikv/mock-tikv/executor.go]: column ID is defined as `id`.

**Pull request merged**